api/routes.js

```diff
@@ -746,13 +747,18 @@ module.exports = (fastify, opt, next) => {
           logger.info("Empty VAST returned", { label: host });
         } else {
           logger.info("Returned VAST and created a session", { label: req.headers['host'], sessionId: session.sessionId });
+          CloudWatchLog("ADS_RETURNED", req.headers['host'], { dur: session.adBreakDuration, session: session.sessionId });
         }
 
         reply.header("Content-Type", "application/xml; charset=utf-8");
         reply.code(200).send(vast_xml);
       }
     } catch (exc) {
-      logger.error(exc, { label: req.headers['host'], sessionId: session.sessionId });
+      if (session) {
+        logger.error(exc, { label: req.headers['host'], sessionId: session.sessionId });
+      } else {
+        logger.error(exc, { label: req.headers['host'] });
```

Comment:
I believe that there are 4 more endpoints that use `session.sessionId` in logger.error(...) 
GET/sessions/:sessionId/tracking
GET/sessions/:sessionId/events
DELETE/sessions/:sessionId
GET/sessions/:sessionId

Reply:
Good point